lib/components/ActiveTextTruncate.vue

```diff
@@ -1,10 +1,13 @@
-<script>
+<script lang="ts">
   import get from 'lodash/get'
   import uniqueId from 'lodash/uniqueId'
   import ResizeObserver from 'resize-observer-polyfill'
+  import { defineComponent } from 'vue'
   import { RequestAnimationFrameWrapper } from '../utils/animation'
 
-  export default {
+  type ActiveTextTruncateData = {textLivePosition:number, resizeObserverKey:string|null}
```

Comment:
Do you have a convention for when you prefer to use a type and when you prefer to use an interface ?

Reply:
I'm moving from using only interfaces (works for everything) to this convention (still learning): https://blog.logrocket.com/types-vs-interfaces-in-typescript/

> Interfaces are basically a way to describe data shapes, for example, an object.
> Type is a definition of a type of data, for example, a union, primitive, intersection, tuple, or any other type.

I'm using interfaces for classes and for objects with multiple implementations. 
Do you have a convention?

Reply:
Then as an object, this would be an interface, no?  :smile: 
But never mind, I was just curious! 

I'm used to starting with interface when describing an object and then turn to a type instead when I need to do an 'operation': use a generic, use utility  types, or use unions, intersections.. It's a bit similar I guess :)